src/App.tsx

```diff
@@ -1,56 +1,92 @@
+    setIsDropdownFocus(false);
+    setAppliedQuerry(newPerson.name.toString());
+  }, []);
+
+  const handleReset = useCallback(() => {
+    setSelectedPerson(null);
+    setQuerry('');
+    setAppliedQuerry('');
+  }, []);
+
+  const handleQueryReset = useCallback(() => {
+    setQuerry('');
+    setAppliedQuerry('');
+  }, []);
+
+  const visablePeople = useMemo(() => {
+    return peopleFromServer
+      .filter(person => person.name
+        .toLowerCase().includes(appliedQuerry.toLowerCase()));
+  }, [appliedQuerry, peopleFromServer]);
+
+  const title = selectedPerson
+    ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+    : 'No matching suggestions';
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {title}
+        <button
+          aria-label="Reset information"
+          onClick={handleReset}
+          type="button"
+          className={cn(
+            'button_delete delete is-medium',
+            { 'is-hidden': !selectedPerson },
+          )}
+        />
       </h1>
 
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
-
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
-        </div>
+      <div
+        className={cn('dropdown', { 'is-active': isDropdownFocus })}
+      >
+        <DropdownForm
+          querry={querry}
+          onQuerry={setQuerry}
+          onAppliedQuerry={setAppliedQuerry}
+          selectedPerson={selectedPerson}
+          onDropdownFocus={setIsDropdownFocus}
+          debounceTimer={debounceTimer}
+        />
+
+        <button
+          aria-label="Close dropdown"
+          onClick={() => {
+            setIsDropdownFocus(false);
+            handleQueryReset();
+          }}
```

Comment:
Move the onClick logic to a separate function. It follows the best practice of separating concerns and keeping your JSX code clean and focused on rendering components rather than defining complex logic directly within it.